theano/sandbox/cuda/tests/test_opt.py

```diff
@@ -91,6 +92,33 @@ def test_local_gpu_contiguous_gpu_contiguous():
                      if isinstance(node.op, basic_ops.GpuContiguous)])
 
 
+def test_local_assert_no_cpu_op():
+    numpy.random.seed(1)
+    m = numpy.random.uniform(-1, 1, (10, 10)).astype("float32")
+    ms = cuda.shared_constructor(m, name="m_shared")
+    out = theano.tensor.tanh(ms**2 + 3)
+
+    mode_local_assert = mode_with_gpu.including("local_assert_no_cpu_op")
+    mode_local_assert = mode_local_assert.excluding("local_gpu_elemwise_1")
+
+    old = config.assert_no_cpu_op
+
+    # If the flag is raise
+    try:
+        config.assert_no_cpu_op = 'raise'
+        assert_raises(AssertionError, theano.function([], out,
+                      mode=mode_local_assert))
```

Comment:
Also, you should not explicitly call `theano.function(...)` from inside assert_raises, but for instance:
```python
    assert_raises(RuntimeError, theano.function, [], out, mode=mode_local_assert)
```